homeassistant/components/calendar/todoist.py

```diff
@@ -496,6 +496,10 @@ def update(self):
             # We had no valid tasks
             return True
 
+        # Make sure the task collection is reset to prevent an
+        # infinite collection repeating the same tasks
+        self.all_project_tasks = []
```

Comment:
why not `self.all_project_tasks.clear()` ?

Reply:
Did not know of it’s existence to be honest. Definitely a lot nicer from a language perspective. Will fix and resubmit after dinner.